### ISSUE TITLE
Bug: React throws many fake errors when warnings are issued

### ISSUE THREAD

**Issue by markboyall** [facebook/react#20882]
In describeNativeComponentFrame, the component rendering is intentionally failed so that we can gather details from the stack trace for some warnings. We have many, many components in our call stack before the warning is triggered, throwing many false errors, making it very difficult for us to use the dev tools to investigate unrelated errors when the warnings are present.

React version: 17.0.1

1. Create a simple minimal React page.
2. Render a simple "hello world" component with a warning.
3. Open the Chrome dev tools, click break on all exceptions with caught exceptions, and then reload the page.

Link to code example:

https://github.com/markboyall/react-break-example

The current behaviour is that it breaks once per component in the stack frame, causing an unmanageable number of fake breaks. The expected behaviour is that React does not interfere with the browser dev tools can be used to debug our own code.

**Comment by bvaughn** [comment]
This isn't a bug. It's the way this feature was designed to work. "Component stacks" are _really_ helpful when debugging errors and warnings in React. The new approach to generating them works without requiring the extra bloat of "debug owner" metadata that we used to have to bundle along with a React application.

The downside is that if you enable "pause on caught exceptions" you may hit a lot of breakpoints. This is also true with the Suspense API (since it works by throwing promises).

This is not something we are likely to change, given the lack of better alternatives.

**Comment by markboyall** [comment]
I'd much rather have extra bloat than have the dev tools not work correctly. Any chance that we can have a setting or something to revert to the previous behaviour? It's hard for me to imagine that "Makes critical browser functionality useless" can possibly ever not be a bug, but if for some reason you disagree, it would be nice to have the option.

**Comment by bvaughn** [comment]
> Any chance that we can have a setting or something to revert to the previous behaviour?

No. I don't think this is the type of thing that we would expose behind a setting. It's core to how React generates its Dev warnings.

**Comment by ncoder** [comment]
I'm also of the opinion that this is an undesirable outcome. I got _very_ confused breaking on errors where my components failed to destructure props.

**Comment by nateabele** [comment]
Agreed. This _really_ sucks. I'd rather not even have a component stack warning.
